dartagnan/src/main/java/com/dat3m/dartagnan/expression/op/IOpUn.java

```diff
@@ -96,9 +98,9 @@ public Formula encode(Formula e, SolverContext ctx) {
     		case MINUS:
     				return bvmgr.subtract(bvmgr.makeBitvector(32, BigInteger.ZERO), bv);
     		case BV2UINT:
-    				return bvmgr.toIntegerFormula(bv, false);
+    				return ARCH_PRECISION > -1 ? bv : bvmgr.toIntegerFormula(bv, false);
```

Comment:
I'm wondering if this code should perform an extension of the provided `bv` instead.
E.g. if you have a `bv8` and convert it to `int` on a 32 bit architecture, it will convert it to ´bv32´ instead.

Reply:
At this point in the code, we don't have information about the precision of the expression (if so, we could just extend `ARCH_PRECISION - prec` bits), thus the best we can do is to convert the expression to an int, and then just create a `bv` with `ARCH_PRECISION`.

Reply:
I think you do have that information via `bvmgr.getLength(bv)`.
So you can extend by `ARCH_PRECISION - bvmgr.getLength(bv)`.

Reply:
Ohhh I was looking to the methods from `BitvectorFormula` instead of the manager. Thanks for the pointer